src/predicate/boxed.rs

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2017, Nick Stevens <nick@bitcurry.com>
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/license/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! Predicate that can wrap other dynamically-called predicates in an
+//! easy-to-manage type.
+
+use std::fmt;
+
+use Predicate;
+
+/// `Predicate` that wraps another `Predicate` as a trait object, allowing
+/// sized storage of predicate types.
+pub struct BoxPredicate<T: ?Sized>(Box<Predicate<Item = T> + Send + Sync>);
+
+impl<T> fmt::Debug for BoxPredicate<T> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.debug_struct("BoxPredicate").finish()
+    }
+}
+
+impl<T> fmt::Display for BoxPredicate<T> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "BoxPredicate")
+    }
+}
```

Comment:
Generally, yes, it would be totally doable by requiring that `T` also implement `Debug`, and then you can print the `Debug` value of `T` as part of the `Debug` impl of `BoxPredicate`. _However_, things get really hairy because of the trait object here. A trait object can only have one trait type, and optionally `Send`/`Sync`.

There are a couple of ways around this. First would be to make the `Predicate` trait subtrait `Debug`. That would then require that anything that implements `Predicate` also implement `Debug`, and that seemed like an annoying restriction.

The other option would be to add an optional `print` method (or something like that with a better name) to the `Predicate` trait. I'm not opposed to this, but I wanted to give it some more thought first.